src/js/tech/tech.js

```diff
@@ -1261,7 +1261,7 @@ Tech.withSourceHandlers = function(_Tech) {
     }
 
     this.sourceHandler_ = sh.handleSource(source, this, this.options_);
-    this.on('dispose', this.disposeSourceHandler);
+    this.one('dispose', this.disposeSourceHandler);
```

Comment:
We want `disposeSourceHandler` itself to be removed on dispose so we need to use `one` instead of `on`. We could also use `off` in `disposeSourceHandler` but I think this is more clear.